src/plugins/data/common/search/search_source/create_search_source.ts

```diff
@@ -56,6 +57,18 @@ export const createSearchSource = (
 ) => async (searchSourceFields: SearchSourceFields = {}) => {
   const fields = { ...searchSourceFields };
 
+  // When we load a saved search and the saved search contains a non index pattern data source this step creates the temperary index patterns and sets the appriopriate query
+  if (
+    searchSourceDependencies.queryStringService &&
```

Comment:
When would `searchSourceDependencies.queryStringService` not be defined? What would we do in such a situation?

Reply:
Spoke to Ashwin; this will never not be defined and this is just a guards.